### DISCUSSION TITLE
Server Component set & delete cookies

### DISCUSSION THREAD

**Discussion by 0-don** [vercel/next.js#49421] [App Router] [answered]
Is there a way to delete or set cookies in server components yet? without a middleware or server actions?

**Comment by leerob** [comment, marked as answer]
You can set cookies in Server Actions and Route Handlers: https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions

**Reply by 0-don** [reply]
so its not posible in sever components yet?

**Reply by icyJoseph** [reply]
Right, is it because rendering has to be pure? or at least, in principle be as close as possible? And setting data as you render is not, pure necessarily?

**Reply by 0-don** [reply]
the problems is that I could set a cookie via serverSideProp previously and with the new app directory its not posible

**Reply by icyJoseph** [reply]
For sure, not having access to the incoming request and outgoing server response certainly has some trade-offs. And I'm not sure if middleware can really replace them all.
